src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -533,57 +533,80 @@ void probe_semi_anti(const RadixContainer<RightType>& radix_container,
 
     jobs.emplace_back(std::make_shared<JobTask>([&, partition_begin, partition_end, current_partition_id]() {
       // Get information from work queue
-      auto& partition = static_cast<Partition<RightType>&>(*radix_container.elements);
+      auto& partition = static_cast<Partition<ProbeColumnType>&>(*radix_probe_column.elements);
 
       PosList pos_list_local;
 
       if (hash_tables[current_partition_id]) {
         // Valid hashtable found, so there is at least one match in this partition
 
         // Accessors are not thread-safe, so we create one evaluator per job
-        MultiPredicateJoinEvaluator multi_predicate_join_evaluator(left, right, secondary_join_predicates);
+        MultiPredicateJoinEvaluator multi_predicate_join_evaluator(build_table, probe_table, mode,
+                                                                   secondary_join_predicates);
 
         for (size_t partition_offset = partition_begin; partition_offset < partition_end; ++partition_offset) {
-          auto& row = partition[partition_offset];
+          auto& probe_column_element = partition[partition_offset];
 
-          if constexpr (retain_null_values) {
```

Comment:
A particular reason to remove the template parameter?

Reply:
It just wasn't needed anymore.

Reply:
I replaced it with a JoinMode template param now.